x-pack/plugins/apm/public/components/routing/templates/apm_service_template/index.tsx

```diff
@@ -265,9 +266,18 @@ function useTabs({ selectedTab }: { selectedTab: Tab['key'] }) {
         path: { serviceName },
         query,
       }),
-      label: i18n.translate('xpack.apm.home.infraTabLabel', {
-        defaultMessage: 'Infrastructure',
-      }),
+      label: (
+        <EuiFlexGroup gutterSize="xs">
+          <EuiFlexItem>
+            {i18n.translate('xpack.apm.home.infraTabLabel', {
+              defaultMessage: 'Infrastructure',
+            })}
+          </EuiFlexItem>
+          <EuiFlexItem>
+            <BetaBadge icon="beaker" />
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      ),
```

Comment:
Can't you add the `BetaBadge` in the `append` property? https://elastic.github.io/eui/#/navigation/tabs